scripts/sync-react.js

```diff
@@ -306,22 +319,29 @@ Or, run this command with no arguments to use the most recently published versio
     )
   }
 
+  const syncPagesRouterReact = pagesRouterReact === null
+  const pagesRouterReactVersion = syncPagesRouterReact
+    ? newVersionStr
+    : pagesRouterReact
   const { sha: baseSha, dateString: baseDateString } = baseVersionInfo
-  for (const fileName of filesReferencingReactPeerDependencyVersion) {
-    const filePath = path.join(cwd, fileName)
-    const previousSource = await fsp.readFile(filePath, 'utf-8')
-    const updatedSource = previousSource.replace(
-      `const nextjsReactPeerVersion = "${baseVersionStr}";`,
-      `const nextjsReactPeerVersion = "${newVersionStr}";`
-    )
-    if (updatedSource === previousSource) {
-      errors.push(
-        new Error(
-          `${fileName}: Failed to update ${baseVersionStr} to ${newVersionStr}. Is this file still referencing the React peer dependency version?`
-        )
+
+  if (syncPagesRouterReact) {
```

Comment:
Was debating to remove this logic since we won't use it for a while. But then it did go through some iterations to get right so I'm also hesitant to remove it and then not be able to restore it easily due to refactor. If it gets annoying, we can remove this logic when it becomes unused i.e. we switch to using stable React in Pages Router.